nservicebus/azure-service-bus/batching.md

```diff
@@ -0,0 +1,72 @@
+
+NOTE: `BrokeredMessage` size is different between [tiers](https://azure.microsoft.com/en-us/documentation/articles/service-bus-premium-messaging/) of Azure Service Bus.
+
+When batching messages sent from a handler, the underlying implementation of batching serializes messages. Serialized batch is usually bigger than the original messages combined. To ensure successful batch sending operation, batch should not exceed the `BrokeredMessage` size. To cater for the overhead caused by serialization, the final batch size is estimated using `MessageSizePaddingPercentage` setting. By default, it's set to 5%. It can be configured using [`MessageSizePaddingPercentage`](/nservicebus/azure-service-bus/configuration/configuration.md#message-senders) configuration of the Message Senders.
+
+By default, messages batch exceeding the maximum allowed size by Azure Service Bus, will throw a [`MessageTooLargeException`](https://msdn.microsoft.com/en-us/library/microsoft.azure.devices.client.exceptions.messagetoolargeexception.aspx). Default behavior can by changed with [`OversizedBrokeredMessageHandler<T>(T)`](/nservicebus/azure-service-bus/configuration/configuration.md#message-senders) configuration of the Message Senders.
+
+
+### Padding and estimated batch size calculation
+
+The following are taken into consideration for batch size calculation:
+
+ * Raw body size in bytes.
+ * Custom headers size (keys and values) in bytes.
+ * Estimated standard properties size in bytes. For string properties, assumed size is 256 bytes.
+ * Additional `MessageSizePaddingPercentage` to account for serialization and internals added by Azure Service Bus library upon sending.
+
+The default value for `MessageSizePaddingPercentage` is 5%. The custom percentage might be required when dealing with consistent size pattern messages. Tables below demonstrate 5% padding and its affect on the message size estimate for various payload sizes.
+
+`BrokeredMessageBodyType` set to `SupportedBrokeredMessageBodyTypes.ByteArray`.
+`MessageSizePaddingPercentage` set to 5%
+
+| Single message payload size   | Size reported by the broker  | Estimated (padded) size | Increase |
+|---|---:|---:|:---:|
+|0K  | 1,426  | 3,088 | 117% |
+|1K   | 2,859 | 4,538 | 59% |
+| 10K  | 15,147 | 17,440 | 16% |
+| 100K  | 138,029 | 146,464 | 7% |
+| 170K  | 233,550 | 246,815 | 6% |
+| 180K  | 247,201 | 261,149 | 6% |
+
+
+`BrokeredMessageBodyType` set to `SupportedBrokeredMessageBodyTypes.Steam`. 
+`MessageSizePaddingPercentage` set to 5%
+
+| Single message payload size   | Size reported by the broker  | Estimated (padded) size | Increase |
+|---|---:|---:|:---:|
+|0K  | 1,401  | 3,101 | 122% |
+|1K   | 2,769 | 4,538 | 64% |
+| 10K  | 15,057 | 17,440 | 16% |
+| 100K  | 137,937 | 146,464 | 7% |
+| 170K  | 233,509 | 246,815 | 6% |
+| 180K  | 247,161 | 261,149  | 6% |
```

Comment:
I've tried to make it as easy to read as possible, but couldn't find anything better... Will leave it as-is and if users ask to clarify, will try to adjust it.